astroplan/periodic.py

```diff
@@ -0,0 +1,295 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+import numpy as np
+import astropy.units as u
+from astropy.time import Time
+
+__all__ = ['PeriodicEvent', 'EclipsingSystem']
+
+
+class PeriodicEvent(object):
+    """
+    A periodic event defined by an epoch and period.
+    """
+    @u.quantity_input(period=u.day, duration=u.day)
+    def __init__(self, epoch, period, duration=None, name=None):
+        """
+
+        Parameters
+        ----------
+        epoch : `~astropy.time.Time`
+            Time of event
+        period : `~astropy.units.Quantity`
+            Period of event
+        duration : `~astropy.units.Quantity` (optional)
+            Duration of event
+        name : str (optional)
+            Name of target/event
+        """
+        self.epoch = epoch
+        self.period = period
+        self.name = name
+        self.duration = duration
+
+    def phase(self, time):
+        """
+        Phase of periodic event, on interval [0, 1). For example, the phase
+        could be an orbital phase for an eclipsing binary system.
+
+        Parameters
+        ----------
+        time : `~astropy.time.Time`
+            Evaluate the phase at this time or times
+
+        Returns
+        -------
+        phase_array : `~numpy.ndarray`
+            Phase at each ``time``, on range [0, 1)
+        """
+        return ((time - self.epoch).to(u.day).value %
+                self.period.to(u.day).value) / self.period.to(u.day).value
+
+
+class EclipsingSystem(PeriodicEvent):
```

Comment:
How do you feel about renaming `EclipsingSystem` to something like `ApproximateEccentricEclipsingSystem`?  (One could imagine splitting that into `EclipsingSystem` as a superclass and `CircularEclipsingSystem` as a subclass, although I'm not sure that's necessary.) 

This might not make sense if your hope is to update the implementation later to be less of an approximation, but my read is that this approximation is intended to be a *permanent* "fast approximation" event.  That leaves room for future slower but more accurate calculations.

Reply:
I am very biased, but I think the majority of eclipsing systems that are observed from the ground are eclipsing binaries and transiting exoplanets, which can be computed with sufficient accuracy for scheduling purposes in this "fast" implementation. I would be thrilled if someone came along and implement a more accurate version someday, but I think that this object will be the workhorse for 90+% of users, so I'm going to preserve the simple name.

Reply:
Hmm.  The hope is that this be not just for "observed from the ground".  But I see your point that it's not necessarily wise to make names that assume future work has already happened...